server/src/main/java/io/druid/server/metrics/QueryCountStatsProvider.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.druid.server.metrics;
+
+public interface QueryCountStatsProvider
```

Comment:
@nishantmonu51 if you are planning to make more implementations it is ok, but otherwise IMO it just makes the codebase more scattered, harder to find the meaning out of it and spawning questions (I asked this question, so probably other people who will read this code will also wonder are there any more implementations planned, try to find them, maybe in extensions, etc.)

Also given the limited usage of QueryCountStatsProvider: only within a single screen in a single class, there is no risk that "someone" will misuse `QueryResource`and call `doPost()` instead of `getSuccessfulQueryCount()`.